internal/pkg/manifest/svc.go

```diff
@@ -238,11 +260,59 @@ type SidecarConfig struct {
 
 // TaskConfig represents the resource boundaries and environment variables for the containers in the task.
 type TaskConfig struct {
-	CPU       *int              `yaml:"cpu"`
-	Memory    *int              `yaml:"memory"`
-	Count     *int              `yaml:"count"` // 0 is a valid value, so we want the default value to be nil.
-	Variables map[string]string `yaml:"variables"`
-	Secrets   map[string]string `yaml:"secrets"`
+	CPU       *int                   `yaml:"cpu"`
+	Memory    *int                   `yaml:"memory"`
+	Count     AutoscalingOrTaskCount `yaml:"count"`
+	Variables map[string]string      `yaml:"variables"`
+	Secrets   map[string]string      `yaml:"secrets"`
+}
+
+// AutoscalingOrTaskCount is a custom type which supports unmarshaling yaml which
+// can either be of type int or type Autoscaling.
+type AutoscalingOrTaskCount struct {
+	TaskCount   *int // 0 is a valid value, so we want the default value to be nil.
+	Autoscaling AutoscalingConfig
+}
+
+// UnmarshalYAML overrides the default YAML unmarshaling logic for the AutoscalingOrTaskCount
+// struct, allowing it to perform more complex unmarshaling behavior.
+// This method implements the yaml.Unmarshaler (v2) interface.
+func (a *AutoscalingOrTaskCount) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	if err := unmarshal(&a.Autoscaling); err != nil {
+		switch err.(type) {
+		case *yaml.TypeError:
+			break
+		default:
+			return err
+		}
+	}
+
+	if !a.Autoscaling.IsEmpty() {
+		return nil
+	}
+
+	if err := unmarshal(&a.TaskCount); err != nil {
+		return errUnmarshalCountOpts
+	}
+	return nil
+}
+
+// AutoscalingConfig represents the configurable options for Auto Scaling.
+type AutoscalingConfig struct {
+	Range        Range   `yaml:"range"`
+	CPU          *string `yaml:"cpu"`
+	Memory       *string `yaml:"memory"`
+	Requests     *int    `yaml:"requests"`
+	ResponseTime *int    `yaml:"response-time"` // TODO: make this to be more humanized like 500ms
```

Comment:
1. Can we do this? I think when the string is something that's parse-able by `time.ParseDuration` then it will parsed properly
```suggestion
	ResponseTime *time.Duration    `yaml:"response_time"` 
```
2. Can we use underscores ("_") instead of dashes "-" to separate words in our manifests? This is consistent with Docker Compose and also our `test_commands` and `start_period` fields.